thefuck/types.py

```diff
@@ -280,6 +280,8 @@ def run(self, old_cmd):
         """
         if self.side_effect:
             compatibility_call(self.side_effect, old_cmd, self.script)
+        if settings.alter_history:
```

Comment:
@nvbn: I thought about adding another `check if Fish` here but then thought this could be decided by the Shell classes implementation. Maybe a comment on `Generic.put_to_history()` would be a good idea, for future reference. What do you think?